client/internal/sender.go

```diff
@@ -62,6 +87,28 @@ func (h *SenderCommon) NextMessage() *NextMessage {
 	return &h.nextMessage
 }
 
+// IsSendingDisabled returns true when onMessage callback is running
+func (h *SenderCommon) IsSendingDisabled() bool {
+	return atomic.LoadInt32(&h.isSendingDisabled) != 0
+}
+
+// DisableScheduleSend temporary preventing ScheduleSend from writing to channel
+func (h *SenderCommon) DisableScheduleSend() {
+
+	atomic.StoreInt32(&h.isSendingDisabled, 1)
+}
+
+// EnableScheduleSend re-enables ScheduleSend and checks if it was called during onMessage callback
```

Comment:
Nit: I would refrain from mentioning onMessage here. It is not a concern of SenderCommon.